packages/apputils-extension/src/notificationplugin.tsx

```diff
@@ -639,19 +641,53 @@ namespace Private {
    */
   let toastify: typeof ReactToastify | null = null;
 
-  function CloseButton(props: CloseButtonProps): JSX.Element {
-    const trans = translator.load('jupyterlab');
+  /**
+   * Interface for CloseButton component
+   */
+  export interface ICloseButtonProps
+    extends React.DetailedHTMLProps<
+      React.ButtonHTMLAttributes<HTMLButtonElement>,
+      HTMLButtonElement
+    > {
+    /**
+     * A function to handle a close event when the CloseButton is clicked
+     */
+    close: (e?: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+    /**
+     * The LabIcon component to be used as the close icon
+     */
+    closeIcon: LabIcon.IReact;
+    /**
+     * Optional boolean to apply margin to the close icon. Default is false.
+     */
+    closeIconMargin?: boolean;
+  }
+
+  export function CloseButton(props: ICloseButtonProps) {
     return (
       <button
-        className={`bp3-button bp3-minimal jp-Button jp-mod-minimal ${TOAST_CLOSE_BUTTON_CLASS}`}
-        title={trans.__('Hide notification')}
-        onClick={props.closeToast}
+        className={`jp-Button jp-mod-minimal ${TOAST_CLOSE_BUTTON_CLASS}${
```

Comment:
You need to keep the bp3 classes.

Reply:
Kept bp3 classes, removed "jp-icon-hover" from props.closeIcon that made both circular and rectangular hover effects appear, see PR description for updated preview